packages/js/src/utils/typescript/tsnode-register.ts

```diff
@@ -1,15 +1,22 @@
-export function tsNodeRegister(file: string, tsConfig?: string) {
+import type { RegisterOptions, Service } from 'ts-node';
+
+export function tsNodeRegister(
+  file: string,
+  tsConfig?: string,
+  tsNodeOptions?: RegisterOptions
+): Service | undefined {
   if (!file?.endsWith('.ts')) return;
   // Register TS compiler lazily
-  require('ts-node').register({
+  const tsNodeService = require('ts-node').register({
     project: tsConfig,
     compilerOptions: {
       module: 'CommonJS',
       types: ['node'],
     },
+    ...tsNodeOptions,
```

Comment:
Do we need to pass `transpileOnly`? I'm not sure if the `swc` version of this type checks.. maybe we should just always `transpileOnly` 🤔 

This also seems kind of separate from this change so I don't know if it should be done in this PR.